llama.cpp

```diff
@@ -3000,6 +3034,22 @@ static void llm_load_hparams(
                     default: model.type = e_model::MODEL_UNKNOWN;
                 }
             } break;
+        case LLM_ARCH_BERT:
+            {
+                ml.get_key(LLM_KV_ATTENTION_LAYERNORM_EPS, hparams.f_norm_eps);
+                ml.get_key(LLM_KV_ATTENTION_CAUSAL, hparams.causal_attn);
+                ml.get_key(LLM_KV_TOKENIZER_TOKEN_TYPE_COUNT, hparams.n_vocab_type);
+
+                switch (hparams.n_embd) {
+                    case 384: // MiniLM
+                        switch (hparams.n_layer) {
+                            case 6:  model.type = e_model::MODEL_22M; break;
+                            case 12: model.type = e_model::MODEL_33M; break;
```

Comment:
Hm, I thought that since it defaults to MODEL_UNKNOWN already that it didn't matter?

https://github.com/ggerganov/llama.cpp/blob/8e6a9d2de0096af7120606c74ee2f26684e87b41/llama.cpp#L1712-L1715

Reply:
Yeah, I think this is consistent with the other models' approach. Added a few more embedding models that work too.

Reply:
Most other models have a `default` case, but if it is already initialized it doesn't matter.